components/webext-storage/src/migration.rs

```diff
@@ -178,18 +212,72 @@ fn read_rows(filename: &Path) -> Vec<LegacyRow> {
         Ok(r) => r,
         Err(e) => {
             log::warn!("Failed to read any rows from the source DB: {}", e);
-            return Vec::new();
+            return (
+                Vec::new(),
+                MigrationInfo {
+                    non_fatal_errors_seen: 1,
+                    open_failure: true,
+                    ..Default::default()
+                },
+            );
         }
     };
+    let rows: Vec<LegacyRow> = rows.filter_map(Result::ok).collect();
+
+    let entries = src_conn.query_row::<i64, _, _>(
```

Comment:
I really think we should try to avoid doing extra reads just for telemetry - it's just adding risk without any value. Can we please try and work out how to learn what you think this will tell you without additional io? ISTM that 30% of this file is now dedicated to telemetry which doesn't seem ideal, so let's KISS where we can.